pkg/server/api.go

```diff
@@ -138,21 +138,10 @@ func (sh *APIHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		w.WriteHeader(http.StatusNotFound)
 		return
 	}
-
-	// the internally saved config should be v3.1, but to eliminate the
-	// potential of race conditions, translate to the requested version
-	// to make sure
+	// we know we're at 3.1 in code.. serve directly, parsing is expensive...
```

Comment:
It's not *just* that parsing is expensive, the core problem is that we're doing it during an HTTP request, and most notably before we write the HTTP headers.

Reply:
I'll add that you're right thanks Colin